src/layer/WmsLayer.ts

```diff
@@ -56,26 +56,30 @@ export class WmsLayer extends AbstractLayer {
       throw new Error('Layer time information is not in ISO8601 format, parsing not supported');
     }
 
-    let result = [];
+    let allTimesUTC = [];
     const times = timeDimension['_'].split(',');
     for (let i = 0; i < times.length; i++) {
       const timeParts = times[i].split('/');
       switch (timeParts.length) {
         case 1:
-          result.push(moment.utc(timeParts[0]).toDate());
+          allTimesUTC.push(moment.utc(timeParts[0]));
           break;
         case 3:
-          const [fromTime, toTime, interval] = timeParts;
-          const intervalDuration = moment.duration(interval);
-          const toTimeMoment = moment.utc(toTime);
-          for (let t = moment.utc(fromTime); t.isSameOrBefore(toTimeMoment); t.add(intervalDuration)) {
-            result.push(t.toDate());
+          const [timePartFromTime, timePartToTime, timePartInterval] = timeParts;
+          const fromTimeMoment = moment.utc(timePartFromTime);
+          const toTimeMoment = moment.utc(timePartToTime);
```

Comment:
Should these two be renamed - do they seem too associated with input params `fromTime` and `toTime`?

Reply:
Renamed